### DISCUSSION TITLE
[Feature] download storage template name instead of uploaded file name

### DISCUSSION THREAD

**Discussion by yarez0** [immich-app/immich#10532] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

using the storage template migration the naming convention is for example : 
{{{album}}}/{{y}}{{MM}}{{dd}}_{{HH}}{{mm}}{{SSS}}

uploading by backup in mobile app name the file IMG_xxxx

the download button in the web ui a file called {{y}}{{MM}}{{dd}}_{{HH}}{{mm}}{{SSS}} download the file but in its original backup file name instead of the storage template name

![image](https://github.com/immich-app/immich/assets/30565737/a2f635ef-c451-4ae4-8da3-796323f22a21)

it would be great to have the possibility of download the file as it is save in the library instead of IMG_xxxx

from discussion https://github.com/immich-app/immich/discussions/10497

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by AlexHoebart-ICPDR** [comment]
I support this request as I think it is very confusing that the name of the download is different to the name of the file on the server.

This is particularly the case when downloading all files of an album. I use a Storage Template and filenames include the datetime, this even works (in most cases) for videos and WhatsApp photos. When I download the album files, the original filenames (from various sources) are used again (IMG_xxx, DSC_xxx, abc4535-...) and therefore the photos cannot be sorted in order of time anymore. I was hoping to overcome exactly this problem with Immich, but the original names in the download defeat the feature of the Storage Templates.

Just a thought: maybe there could be "Download Templates" for renaming files during download to keep this flexible - either to use the original file names, the same as in the Storate Templates or even some other format.

Thanks for this great app, looking forward to further improvements.

**Comment by carlosjfcasero** [comment]
+1. I would love to see this feature

**Comment by xpac1985** [comment]
Adding a backlink to my issue https://github.com/immich-app/immich/issues/18830 here, because this has existed as a feature/behavior at some point before, it seems - or at least was planned.

**Comment by stickflowerfightclub** [comment]
Hi, any news about this request please?
When I download a picture from Immich, I have the same problem: the picture is downloaded with its original name (e.g., whatsapp_img_3232.jpg) instead of the filename template I set up.

My Immich server is up to date of course (v2.3.1).

Many thanks in advance !

**Comment by clamel77** [comment]
I also would love this feature

**Comment by tangles-0** [comment]
+1 for this feature for privacy reasons

**Comment by b4stOss** [comment]
I would also like to have this feature because the fact that the file names are the original ones means that downloaded photos are no longer necessarily in order, even though they are sorted correctly on Immich. This is really inconvenient when you want to download an album to work on it locally on your PC (which is my case here). I found this [MR](https://github.com/immich-app/immich/pull/24472) that solved the problem, but it has been closed unmerged

**Comment by clamel77** [comment]
I also support this feature, very useful for me
